crypto/threads_pthread.c

```diff
@@ -292,29 +294,27 @@ static struct rcu_qp *get_hold_current_qp(struct rcu_lock_st *lock)
 static void ossl_rcu_free_local_data(void *arg)
 {
     OSSL_LIB_CTX *ctx = arg;
-    CRYPTO_THREAD_LOCAL *lkey = ossl_lib_ctx_get_rcukey(ctx);
-    struct rcu_thr_data *data = CRYPTO_THREAD_get_local(lkey);
+    struct rcu_thr_data *data = CRYPTO_THREAD_get_local_ex(CRYPTO_THREAD_LOCAL_RCU_KEY, ctx);
 
     OPENSSL_free(data);
-    CRYPTO_THREAD_set_local(lkey, NULL);
+    CRYPTO_THREAD_set_local_ex(CRYPTO_THREAD_LOCAL_RCU_KEY, ctx, NULL);
```

Comment:
Can you swap lines 300 and 299? so `OPENSSL_free()` follows `_set_local_ex(NULL)`?
thanks.